plugin-server/src/worker/plugins/run.ts

```diff
@@ -443,3 +429,58 @@ async function filterPluginMethods<T>(
 
     return filteredList
 }
+
+async function formatConfigTemplates(
+    hub: Hub,
+    pluginConfig: PluginConfig,
+    event: PostIngestionEvent
+): Promise<Record<string, any>> {
+    const team = await hub.teamManager.fetchTeam(event.teamId)
+    if (!team) {
+        throw new Error('Team not found')
+    }
+
+    const schema = pluginConfig.plugin?.config_schema
+    if (!schema) {
+        // NOTE: This shouldn't be possible and is more about typings
+        return pluginConfig.config
+    }
+
+    const schemaObject: Record<string, PluginConfigSchema> = Array.isArray(schema)
+        ? Object.fromEntries(schema.map((field) => [field.key, field]))
+        : schema
+
+    const webhookFormatter = new MessageFormatter({
+        event,
+        team,
+        siteUrl: hub.SITE_URL || 'http://localhost:8000',
+        // TODO: What about pluginConfig.name ?
+        sourceName: pluginConfig.plugin?.name || 'Unnamed plugin',
```

Comment:
we should use `pluginConfig.name` that's what the user chose to name it - e.g. when they create 2 generic webhook instances we'd want to see what the user named it instead of `Generic Webhook` they can then also more easily find it in the UI as we operate on `pluginConfig.name` not `plugin.name` (which we did back in the day when you could have only 1)